metadata_backend/api/handlers/api_handlers.py

```diff
@@ -0,0 +1,458 @@
+
+        :param req: DELETE request
+        :raises: HTTPUnauthorized if folder published
+        :raises: HTTPUnprocessableEntity if template does not belong to current user
+        :returns: HTTPNoContent response
+        """
+        schema_type = req.match_info["schema"]
+        self._check_schema_exists(schema_type)
+        collection = f"template-{schema_type}"
+
+        accession_id = req.match_info["accessionId"]
+        db_client = req.app["db_client"]
+
+        await Operator(db_client).check_exists(collection, accession_id)
+
+        await self._handle_check_ownedby_user(req, collection, accession_id)
+
+        user_op = UserOperator(db_client)
+        current_user = get_session(req)["user_info"]
+        check_user = await user_op.check_user_has_doc(collection, current_user, accession_id)
+        if check_user:
+            await user_op.remove_objects(current_user, "templates", [accession_id])
+        else:
+            reason = "This template does not seem to belong to any user."
+            LOG.error(reason)
+            raise web.HTTPUnprocessableEntity(reason=reason)
+
+        accession_id = await Operator(db_client).delete_metadata_object(collection, accession_id)
+
+        LOG.info(f"DELETE template with accession ID {accession_id} in schema {collection} was successful.")
+        return web.Response(status=204)
+
+
+class StaticHandler:
+    """Handler for static routes, mostly frontend and 404."""
+
+    def __init__(self, frontend_static_files: Path) -> None:
+        """Initialize path to frontend static files folder."""
+        self.path = frontend_static_files
+
+    async def frontend(self, req: Request) -> Response:
+        """Serve requests related to frontend SPA.
+
+        :param req: GET request
+        :returns: Response containing frontpage static file
+        """
+        serve_path = self.path.joinpath("./" + req.path)
+
+        if not serve_path.exists() or not serve_path.is_file():
+            LOG.debug(f"{serve_path} was not found or is not a file - serving index.html")
+            serve_path = self.path.joinpath("./index.html")
+
+        LOG.debug(f"Serve Frontend SPA {req.path} by {serve_path}.")
+
+        mime_type = mimetypes.guess_type(serve_path.as_posix())
+
+        return Response(body=serve_path.read_bytes(), content_type=(mime_type[0] or "text/html"))
+
+    def setup_static(self) -> Path:
+        """Set path for static js files and correct return mimetypes.
+
+        :returns: Path to static js files folder
+        """
+        mimetypes.init()
+        mimetypes.types_map[".js"] = "application/javascript"
+        mimetypes.types_map[".js.map"] = "application/json"
+        mimetypes.types_map[".svg"] = "image/svg+xml"
+        mimetypes.types_map[".css"] = "text/css"
+        mimetypes.types_map[".css.map"] = "application/json"
+        LOG.debug("static paths for SPA set.")
+        return self.path / "static"
+
+
+# Private functions shared between handlers
```

Comment:
for this i think we can create a separate file `common.py` in handlers submodule, makes the handlers cleaner and we will most likely add more. e.g. I will add more things with CSV PR.